packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -125,11 +125,14 @@ export class EngineReportingExtension<TContext = any>
       for (const [key, value] of o.request.headers) {
         if (
           this.options.privateHeaders &&
-          typeof this.options.privateHeaders === 'object' &&
+          Array.isArray(this.options.privateHeaders) &&
           // We assume that most users only have a few private headers, or will
           // just set privateHeaders to true; we can change this linear-time
           // operation if it causes real performance issues.
-          this.options.privateHeaders.includes(key.toLowerCase())
+          this.options.privateHeaders.some(privateHeader => {
+            // Headers are case-insensitive, and should be compared as such.
+            return privateHeader.toLowerCase() === key.toLowerCase();
```

Comment:
This makes sense to me! We can do things to improve performance of course, like not calling `key.toLowerCase()` for every `privateHeader`, making sure to store private headers in lower case, or even using something like a `Set` for easy lookup (which I think is what that comment also alludes to), but none of that is likely to matter in practice, and this makes the intention immediately clear.

Reply:
Agree.  I tried to avoid doing any micro-optimization right now since we'll (probably soon-ish?) want to introduce the ability to define this as a function and its less clear how beneficial it would be to build that out given that direction (and such an implementation could benefit from its own optimizations).  Unless there are a large number of `privateHeaders` defined here, I don't think this relatively simple string manipulation would benefit, noticeably, right now.